### DISCUSSION TITLE
Correct way to resolve URL for given page file?

### DISCUSSION THREAD

**Discussion by 0x241F31** [vikejs/vike#680] [Discussion]
Hey. I want to make table of content to my blog, so I'm using `import.meta.glob` like this:

`index.page.server.tsx`:
```tsx
const articles = import.meta.glob('./*.md', { eager: true })

// ctx is empty here btw
export function Page(ctx) {
   return (
      <>
         <h3>Articles</h3>
         <ul>
            {Object.entries(articles).map(([fn, module]) => (
               <li>
                  {module.date} <a href={'/blog/' + fn.split('.page')[0]}>{module.title}</a>
               </li>
            ))}
         </ul>
      </>
   )
}
```
Is there a more convinient way to get url for a page? 
I tried with `as: 'url'` in glob function, but it returns `{ './hi.page.md': [Function: MDXContent] }`, so it's useless with VPS.

I was able to access `PageContextBuiltIn` in page, but can't see how it can help me. 

https://github.com/brillout/vite-plugin-ssr/issues/49 - related?

**Comment by brillout** [comment]
For improved dev speed, since Vite does on-demand transpiling, I'd recommend using something like the following instead of `import.meta.glob('/pages/*.md', { eager: true })`: https://github.com/brillout/vite-plugin-ssr/blob/fef733e46fc96297be2ce08821677777acc053fd/docs/headings.tsx#L206-L224 Yes, it's slightly repetitive, but it's very well worth it if you have many pages.
